packages/upload-core/src/upload.js

```diff
@@ -267,6 +267,20 @@ class Upload {
         return false;
       }
     }
+    if (this.options.allowedFileNameCharacters) {
+      const fileName = this.file.name.substring(
+        0,
+        this.file.name.lastIndexOf('.')
+      );
+      const regExp = new RegExp(
+        `([^${this.options.allowedFileNameCharacters}])`,
```

Comment:
Would it be more powerful to let the user define their own regex? They could do more to ensure it matches a predetermined formatted string or something. We can default it to `/[^_a-zA-Z0-9 ]/` which would be the same default behavior.